libs/partners/mongodb-atlas/langchain_mongodb_atlas/vectorstores.py

```diff
@@ -0,0 +1,473 @@
+from __future__ import annotations
+
+import logging
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    Generator,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+    TypeVar,
+    Union,
+)
+
+import numpy as np
+from langchain_community.vectorstores.utils import maximal_marginal_relevance
+from langchain_core.documents import Document
+from langchain_core.embeddings import Embeddings
+from langchain_core.runnables.config import run_in_executor
+from langchain_core.vectorstores import VectorStore
+
+if TYPE_CHECKING:
+    from pymongo.collection import Collection
+
+MongoDBDocumentType = TypeVar("MongoDBDocumentType", bound=Dict[str, Any])
+VST = TypeVar("VST", bound=VectorStore)
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_INSERT_BATCH_SIZE = 1000
```

Comment:
subtle increase in `DEFAULT_INSERT_BATCH_SIZE` will note in the PR.
100 --> 1000

Reply:
I'd prefer these kind of behavior changes to happen separately so they don't get lost in "refactor" commits.

Reply:
Good point. I'll take it out.

Reply:
Removed and encapsulated this follow-up [in a ticket](https://jira.mongodb.org/browse/PYTHON-4222).